atlite/config.py

```diff
@@ -1,146 +1,199 @@
-
-    # Notify user of empty config
-    if not config_path:
-        logger.warning("No valid configuration file found in default and home directories. "
-                       "No configuration is loaded, manual configuration required.")
-
-def _update_variables():
-    """Update list of provided attributes by the module."""
-
-    global ATTRS
-
-    ATTRS = {k for k,v in globals().items() if not k.startswith("_") and not callable(v)}
-
-    # Manually remove imported modules and the attribute itself from the list
-    ATTRS = ATTRS - {"ATTRS", "logging",
-                     "logger", "os", "pkg_resources", "yaml"}
-
-_update_hooks.append(_update_variables)
-
-
-# Load the configuration at first module import
-reset()
+def _make_property(var, ispath=True):
+    if ispath:
+        def getvar(self):
+            return self.resolve_filepath(getattr(self, f"_{var}"))
+    else:
+        def getvar(self):
+            return getattr(self, f"_{var}")
+    def setvar(self):
+        setattr(self, f"_{var}")
+        self.update_hooks
+    # Maybe add link to online documentation to doc-string
+    return property(getvar, setvar, doc=f"Get or set config variable {var}")
+
+def add_properties(cls):
+    for var in cls._VARIABLES:
+        setattr(cls, var, _make_property(var))
+    return cls
+
+@add_properties
+class Config:
+    _VARIABLES = ["config_path", "cutout_dir", "sarah_dir", "windturbine_dir",
+                  "solarpanel_dir", "gebco_path"]
+
+    _FILE_NAME = Path(".atlite.config.yaml")
+    _FILE_SEARCH_PATH = Path.home().joinpath(_FILE_NAME)
+    _DEFAULT_FILE_NAME = Path("config.default.yaml")
+    _DEFAULT_SEARCH_PATH = Path(pkg_resources.resource_filename("atlite", str(_DEFAULT_FILE_NAME)))
+
+    # Functions called on config change
+    _UPDATE_HOOKS = []
+
+    @staticmethod
+    def from_dict(config_dict):
+        return Config(config_dict=config_dict)
+
+    @staticmethod
+    def from_file(config_path):
+        return Config(config_path=config_path)
+
+    def __init__(self, config_dict=None, config_path=None):
+        """Create a config object using a dictionary or by specifying a config file path.
+
+        If neither `config_dict` nor `config_path` are given, then try to read a config
+        file from one of the default locations.
+
+        Parameters
+        ----------
+        config_dict : dict
+            Dictionary containing the values for a new config, takes priority
+            over reading configurations from file.
+            (Default: None)
+        config_path : string or os.path
+            Full path of a valid config file (including arbitrary name).
+            (Default: None)
+        """
+
+        for var in self._VARIABLES:
+            setattr(self, f"_{var}", None)
+
+        # Prefer config explicitly provided as a dict
+        if isinstance(config_dict, dict):
+            logger.info(f"Loading config from dict.")
+            config_dict.setdefault("config_path", ".")
```

Comment:
```suggestion
            config_dict.setdefault("config_path", _FILE_SEARCH_NAME)
```

For consistency, let's refer with `config_path` to the path of the configuration file, not its parent directory.